aggregator/pom.xml

```diff
@@ -100,9 +100,27 @@
                             <pattern>org.apache.hadoop.hive.</pattern>
                             <shadedPattern>${rapids.shade.package}.hadoop.hive.</shadedPattern>
                             <excludes>
+                                <!-- FIXME
+                                    Seems we need to exclude all the public classes being used
+                                    directly and indirectly by plugin to support row-based Hive
+                                    UDF, otherwise it will get a ClassNotFoundException as below:
+                                        E Caused by: java.lang.ClassNotFoundException: com.nvidia.shaded.spark.hadoop.hive.serde2.objectinspector.ObjectInspector
+                                        E     at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
+                                    Or, just simply remove this relocation.
```

Comment:
We need to shade hive as part of shading ORC (we need to shade it and all of its dependencies, or we risk the hive jar either missing in the Spark distribution at runtime or conflicting with the version that the ORC version we're shading needs).

I think we only need to limit the hive relocations to the hive-storage-api classes.  Unfortunately there isn't a nice package prefix that would select only those classes.  We may need to split up the input/output classes that need ORC (and thus the relocations via shading) from the UDF classes that just need the aggregation part of the shading (i.e.: no relocations).  Separating out the Hive-related UDF classes and/or the ORC input/output classes into a separate Maven project would give us a way to control the shading on a more granular basis.

Reply:
Thanks Jason. Good to know.
Is this change here OK for now ?

Reply:
I think what's here is quite fragile, but I guess it's OK for now.  However we should prioritize #1398 to fix this in a better way.